src/prsdk/persistence/serializers/neural_network_serializer.py

```diff
@@ -0,0 +1,83 @@
+"""
+Serializer for the Neural Network Predictor class.
+"""
+import json
+from pathlib import Path
+
+import joblib
+import torch
+
+from data.cao_mapping import CAOMapping
+from persistence.serializers.serializer import Serializer
+from predictors.neural_network.torch_neural_net import TorchNeuralNet
+from predictors.neural_network.neural_net_predictor import NeuralNetPredictor
+
+
+class NeuralNetSerializer(Serializer):
+    """
+    Serializer for the NeuralNetPredictor.
+    Saves config necessary to recreate the model, the model itself, and the scaler for the data to a folder.
+    """
+    def save(self, model: NeuralNetPredictor, path: Path):
+        """
+        Saves model, config, and scaler into format for loading.
+        Generates path to folder if it does not exist.
+        :param model: the neural network predictor to save.
+        :param path: path to folder to save model files.
+        """
+        if model.model is None:
+            raise ValueError("Model not fitted yet.")
+        path.mkdir(parents=True, exist_ok=True)
+
+        config = {
+            "context": model.cao.context,
+            "actions": model.cao.actions,
+            "outcomes": model.cao.outcomes,
+            "features": model.features,
+            "label": model.label,
+            "hidden_sizes": model.hidden_sizes,
+            "linear_skip": model.linear_skip,
+            "dropout": model.dropout,
+            "device": model.device,
+            "epochs": model.epochs,
+            "batch_size": model.batch_size,
+            "optim_params": model.optim_params,
+            "train_pct": model.train_pct,
+            "step_lr_params": model.step_lr_params
+        }
+        with open(path / "config.json", "w", encoding="utf-8") as file:
+            json.dump(config, file)
+        # Put model on CPU before saving
+        model.model.to("cpu")
```

Comment:
Move to CPU so that we don't error if we go from M1 to NVIDIA or change architectures like that